src/api/block_builder/input_selection/helpers.rs

```diff
@@ -0,0 +1,185 @@
+                            *unlock_address.nft_id()
+                                == nft_output
+                                    .nft_id()
+                                    .or_from_output_id(input_signing_data.output_id().expect("Invalid output id"))
+                        } else {
+                            false
+                        }
+                    }
+                    _ => false,
+                },
+                _ => false,
+            }
+        }) {
+            Some(position) => {
+                // Insert after the output we need
+                sorted_inputs.insert(position + 1, input.clone());
+            }
+            None => {
+                // insert before address
+                let alias_or_nft_address = match &input.output {
+                    Output::Alias(alias_output) => Some(Address::Alias(AliasAddress::new(
+                        alias_output
+                            .alias_id()
+                            .or_from_output_id(input.output_id().expect("Invalid output id")),
+                    ))),
+                    Output::Nft(nft_output) => Some(Address::Nft(NftAddress::new(
+                        nft_output
+                            .nft_id()
+                            .or_from_output_id(input.output_id().expect("Invalid output id")),
+                    ))),
+                    _ => None,
+                };
+
+                if let Some(alias_or_nft_address) = alias_or_nft_address {
+                    // Check for existing outputs for this address, and insert before
+                    match sorted_inputs.iter().position(|input_signing_data| {
+                        Address::try_from_bech32(&input_signing_data.bech32_address)
+                            .expect("Safe to unwrap, we encoded it before")
+                            .1
+                            == alias_or_nft_address
+                    }) {
+                        Some(position) => {
+                            // Insert before the output with this address required for unlocking
+                            sorted_inputs.insert(position, input.clone());
+                        }
+                        // just push output
+                        None => sorted_inputs.push(input.clone()),
+                    }
+                } else {
+                    // just push basic or foundry output
+                    sorted_inputs.push(input.clone());
+                }
+            }
+        }
+    }
+
+    Ok(sorted_inputs)
+}
+
+// Check if an address is required for unlockig an output in any unlock condition
+pub(crate) fn output_contains_address(output: &Output, address: &Address, current_time: u32) -> bool {
+    if let Some(unlock_conditions) = output.unlock_conditions() {
+        if let Some(UnlockCondition::Address(address_unlock_condition)) =
+            unlock_conditions.get(AddressUnlockCondition::KIND)
```

Comment:
Use the convenience methods instead of .get ?

Reply:
Will do, so confusing with all these methods